operator/builtin/input/syslog/syslog_test.go

```diff
@@ -80,18 +80,46 @@ func SyslogInputTest(t *testing.T, cfg *SyslogInputConfig, tc syslog.Case) {
 
 func NewSyslogInputConfigWithTcp(syslogCfg *syslog.SyslogParserConfig) *SyslogInputConfig {
 	cfg := NewSyslogInputConfig("test_syslog")
+	cfg.SyslogParserConfig = *syslogCfg
 	cfg.Tcp = tcp.NewTCPInputConfig("test_syslog_tcp")
 	cfg.Tcp.ListenAddress = ":14201"
 	cfg.OutputIDs = []string{"fake"}
-	cfg.Syslog = syslogCfg
 	return cfg
 }
 
 func NewSyslogInputConfigWithUdp(syslogCfg *syslog.SyslogParserConfig) *SyslogInputConfig {
 	cfg := NewSyslogInputConfig("test_syslog")
+	cfg.SyslogParserConfig = *syslogCfg
 	cfg.Udp = udp.NewUDPInputConfig("test_syslog_udp")
 	cfg.Udp.ListenAddress = ":12032"
 	cfg.OutputIDs = []string{"fake"}
-	cfg.Syslog = syslogCfg
 	return cfg
 }
+
+func TestConfigYamlUnmarshal(t *testing.T) {
+	base := `type: syslog_input
+protocol: utc5404
```

Comment:
nit: It doesn't matter for the test, but this should really use an accepted value such as `rfc3164` or `rfc5424`.